### ISSUE TITLE
Ubuntu 20.04 problems with gdm and Gnome if console login required

### ISSUE THREAD

**Issue by matt335672** [neutrinolabs/xrdp#1565]
This issue has been raised as a placeholder, to which more information can be added as it becomes available.

I've spent some time trying to get 20.04 LTS with GNOME working with XRDP. Config is fairly standard:-

- xrdp and xorgxrdp installed.
- dbus-user-session removed (see the [wiki](../wiki/Debian-dbus-user-session-package))

To date, there are two significant issues as follows:

1. If an XRDP session is created, the same user cannot log in on the console using gdm3 using either Xorg or Xwayland.
2. If the first issue is worked around, the same user can log in over XRDP and on the console. However, the two GNOME sessions appear to be coupled in some way, despite having different values for `DBUS_SESSION_BUS_ADDRESS`.

The simplest workaround I've found for the first issue involves rebuilding the gdm3 source package and back-porting commit GNOME/gdm@32646105196ed7a687f3684b69ea544dd01d2ade.

The second issue can be demonstrated as follows:-
- Log in an XRDP session
- Log in a console session (Xorg)  with the same user
- Lock both sessions

At this point, attempting to unlock the console has no effect, but unlocking the XRDP session unlocks both sessions.

This problem seems to be related to `gnome-shell` rather than `gdm`. Running `gdm` under debug shows that the correct unlock message is sent to `systemd-logind` over DBUS. Using `loginctl` to unlock the session also shows the same behaviour.

Things get even stranger if the console is logged in on Wayland. Attempting to start the first terminal window on the XRDP session results in the terminal appearing on the Wayland console with the `DISPLAY` and `DBUS_SESSION_BUS_ADDRESS` set pointing to the XRDP session!

In short, if you're trying to use GNOME on 20.04 with XRDP, there are significant limitations at present. Other desktop environments do not appear to have these problems.

**Comment by mwezdeck** [comment]
Hi,

I installed xrdp on Ubuntu 20.04 with gdm. 
I solved the issue with black screen after logging via RDP.

However, if the machine has locked screen, then I can't log in again.
The workaround, which I tested and which works for me is as follows:
1. connect through remote desktop Windows 10 app to your Ubuntu
2. Try to unlock
3. If unlock is not successfull, then open VNC client and connect to your machine.
4. VNC: unlock your desktop through VNC
5. XRDP: XRDP should unlock your desktop through xrdp, too

Enjoy!

Is there a possibility to assign somebody from dev team to resolve this issue? Workaround is anoying a little bit.

**Comment by matt335672** [comment]
Hi @mwezdeck 

Thanks for the workaround.

When you say "unlock is not successful", do you mean you can type OK, but your password is not accepted? I'm wondering if this is a keyboard mapping issue you're experiencing.

On my GDM unlock screen there's a small button with an eye on it. If you press this you can see what you're typing:-

![capture](https://user-images.githubusercontent.com/30179339/125935773-a9df597a-6a49-47b8-9ab1-43a0a29ac85e.png)

The next time this happens, can you press it and see if your password is being entered correctly?

Thanks.

**Comment by matt335672** [comment]
I'm closing this now. I think the root causes of this are now well understood.
